library/mps_writer.c

```diff
@@ -0,0 +1,424 @@
+/*
+ *  Message Processing Stack, Reader implementation
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of Mbed TLS (https://tls.mbed.org)
+ */
+
+#include "common.h"
+
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+
+#include "mps_writer.h"
+#include "mps_trace.h"
+
+#if ( defined(__ARMCC_VERSION) || defined(_MSC_VER) ) && \
+    !defined(inline) && !defined(__cplusplus)
+#define inline __inline
+#endif
+
+#if defined(MBEDTLS_MPS_ENABLE_TRACE)
+static int mbedtls_mps_trace_id = MBEDTLS_MPS_TRACE_BIT_WRITER;
+#endif /* MBEDTLS_MPS_ENABLE_TRACE */
+
+#define MIN( x, y ) ( ( x ) < ( y ) ? ( x ) : ( y ) )
+
+#include <string.h>
+
+void mbedtls_mps_writer_init( mbedtls_mps_writer *wr,
+                              unsigned char *queue,
+                              mbedtls_mps_size_t queue_len )
+{
+    mbedtls_mps_writer dst =  { .state = MBEDTLS_MPS_WRITER_PROVIDING,
+                                .out   = NULL,
+                                .queue = queue,
+                                .out_len   = 0,
+                                .queue_len = queue_len,
+                                .committed = 0,
+                                .end       = 0,
+                                .queue_next      = 0,
+                                .queue_remaining = 0 };
+
+    *wr = dst;
```

Comment:
Thanks, I didn't know that. Since it doesn't make a difference to the compiler or readability, I prefer not to try out whether VST likes it or not, and leave it as is.